Tests/CSharp/CodeRefactorings/ContractEnsuresNotNullReturnTests.cs

```diff
@@ -0,0 +1,97 @@
+using System;
+using NUnit.Framework;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis;
+using System.Collections.Immutable;
+using RefactoringEssentials.CSharp.CodeRefactorings;
+using System.Diagnostics.Contracts;
+
+namespace RefactoringEssentials.Tests.CSharp.CodeRefactorings
+{
+    [TestFixture]
+    public class ContractEnsuresNotNullReturnTests : CSharpCodeRefactoringTestBase
+    {
+        public int? play()
+        {
+            Contract.Ensures(Contract.Result<int?>() != null);
+            return null;
+        }
+
+        public ContractEnsuresNotNullReturnCodeRefactoringProvider play2()
+        {
+            return null;
+        }
+
+        [Test]
+        public void ValueTypeReturnType()
+        {
+            TestWrongContext<ContractEnsuresNotNullReturnCodeRefactoringProvider>(
+                @"class Test
+            {
+                public $int Foo() 
+                { 
+                }
+            }");
+        }
+
+        [Test]
+        public void NullableValueTypeLocalVariable()
+        {
+            TestWrongContext<ContractEnsuresNotNullReturnCodeRefactoringProvider>(
+                @"class Test
+            {
+                public void Foo() 
+                { 
+                    $int? foo;
+                }
+            }");
+        }
+
+        [Test]
+        // Not sure why anyone would want this when can just change the return type to be non nullable. Maybe you have to implement an interface you don't control or something strange like that.
+        public void NullableValueTypeReturnType()
+        {
+            Test<ContractEnsuresNotNullReturnCodeRefactoringProvider>(
+                @"class Test
+{
+    $public int? Foo() 
+    { 
+    }
+}", @"class Test
+{
+    public int? Foo() 
+    {
+        Contract.Ensures(Contract.Result<int?>() != null);
+    }
+}");
+        }
+
+        [Test]
+        public void ObjectReturnType()
+        {
+            Test<ContractEnsuresNotNullReturnCodeRefactoringProvider>(
+                @"class Test
+{
+    $public Cedd Foo() 
+    { 
+    }
+}", @"class Test
+{
+    public Cedd Foo() 
+    {
+        Contract.Ensures(Contract.Result<Cedd>() != null);
+    }
+}");
+        }
+
```

Comment:
Again, sorry about that, this was due to bad merging. 

Methods with a void return type aren't relevant I think (they don't return anything, so impossible to have a contract on the return value). Property getters / indexers are a good idea, but I might do this as a separate class / commit if that's ok (it will have to inherit from a different base I think - the current one requires a MethodDeclaration (SpecializedCodeRefactoringProvider<MethodDeclarationSyntax>).

I am less excited by Lambda's and delegates, but I can have a go at adding them if you like. I imagine that the inheritance might be an issue again with SpecializedCodeRefactoringProvider<MethodDeclarationSyntax> so I would have to look in to that and potentially create more classes.

Cheers

Cedd